services/ingestion/etl/fts/xls/src/lib/transform.js

```diff
@@ -0,0 +1,234 @@
+ * - `Total amount`
+ * - `Funding Type`
+ * - `Type`
+ *
+ * @memberof FtsXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Budget}
+ */
+const getBudget = record => {
+  const fundingArea = [];
+  // Use `Type` field if provided.
+  if (record.Type) {
+    fundingArea.push(record.Type);
+  }
+
+  // Use `Funding Type` field if provided.
+  if (record['Funding Type']) {
+    fundingArea.push(record['Funding Type']);
+  }
+
+  const budget = {
+    eu_contrib: sanitizeBudgetItem({
+      value: record['Total amount'],
+      currency: 'EUR',
+      raw: record['Total amount'],
+    }),
+    funding_area: fundingArea,
+    mmf_heading: '',
+    other_contrib: sanitizeBudgetItem(),
+    private_fund: sanitizeBudgetItem(),
+    public_fund: sanitizeBudgetItem(),
+    total_cost: sanitizeBudgetItem(),
+  };
+
+  return budget;
+};
+
+/**
+ * Preprocess project_locations
+ * Input fields taken from the `record` are:
+ * - `Address`
+ * - `City`
+ * - `Postal code`
+ * - `Country / Territory`
+ * - `NUTS2`
+ * - `Geographical Zone`
+ *
+ * @memberof FtsXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array} List of {Location} objects for `project_locations` field
+ */
+const getLocations = record => {
+  const locations = [];
+  const addressItems = record.Address.split(';');
+  const postalCodes = record['Postal code'].split(';');
+  const cities = record.City.split(';');
+  const items = record['Country / Territory'].split(';');
+
+  items.forEach((item, key) => {
+    // Get country code from country name
+    const countryCode = getCodeByCountry(item);
+
+    const location = {
+      address: addressItems[key],
+      centroid: null,
+      country_code: getCountryCode(countryCode),
+      location: null,
+      nuts: [],
+      postal_code: postalCodes[key],
+      region: '',
+      town: cities[key],
+    };
+
+    // It's possible that some items have this.
+    // If present, use it.
+    if (item.NUTS2) {
+      location.nuts.push({
+        code: item.NUTS2,
+        name: '',
+        level: getNutsCodeLevel(item.NUTS2),
+        year: null,
+      });
+    }
+
+    locations.push(location);
+  });
+
+  return locations;
+};
+
+/**
+ * Preprocess third_parties
+ *
+ * Input fields taken from the `record` are:
+ * - `Name of beneficiary`
+ * - `Coordinator`
+ * - `Address`
+ * - `Country / Territory`
+ *
+ * @memberof FtsXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array} List of {ThirdParty} objects for `third_parties` field
+ */
+const getThirdParties = record => {
+  const actors = [];
+  const coordinators = record.Coordinator.split(';');
+  const items = record['Name of beneficiary'].split(';');
+  const addressItems = record.Address.split(';');
+  const countryItems = record['Country / Territory'].split(';');
+
+  items.forEach((item, key) => {
+    let role = 'participant';
```

Comment:
Isn't it a `beneficiary`?

Reply:
Good question, initially I used https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/cordis/csv/src/lib/transform.js#L166 for the role and now checking others such as https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/home/xls/src/lib/transform.js see that the `type` is `beneficiary` so I updated it to include the `beneficiary` on type to match the other ETLs.

Reply:
damn, this is so confusing...